### ISSUE TITLE
Wrong order of highlights in text with columns. 

### ISSUE THREAD

**Issue by ichnilatis-gr** [koreader/koreader#12114]
* KOReader version: 2024.04-209
* Device: PB 628

#### Issue
When we make highlights in a text with columns, the order of the highlights in the annotation list is not their order in the text, but the order they apear in a vertical scan of the whole page.
I give you an example. Note that the highlights in the list are set for reverse display.

![Reader_2024-06-27 (2) (Medium)](https://github.com/koreader/koreader/assets/76203123/c0ef930e-879c-4fb1-a99e-00cde50c2c6a)

![Reader_2024-06-27 (Medium)](https://github.com/koreader/koreader/assets/76203123/2b0ca887-11e7-4092-9aff-ec051e04602d)

Could it be fixed so that the highlights are displayed in the correct order in the annotation list? 
Thank you!

**Comment by hius07** [comment]
Assuming it's a pdf, I'm afraid it is difficult to fix.
To determine the order we compare y (vertical coordinate) of the highlighted boxes. y of 3 is less than y of 2, so 3 goes before 2.
A good idea how to do this for 2 columns is welcome.

**Comment by mergen3107** [comment]
Just a thought. 
Does reflow mode detect columns? From kopt2pdf, for example.

**Comment by ichnilatis-gr** [comment]
The same problem exists even in reflow mode.

**Comment by poire-z** [comment]
> A good idea how to do this for 2 columns is welcome.

I don't think we can even detect a page is 2-columns, and not 3 columns or a mix layout (part of the page full width, some parts 2 columns - or the main text taking 2/3 of the width with some footnotes in the remaining 1/3...)
The only thing possibly trustable (but there's no garantee, PDF rendering software may absolutely position all the words without any required order?) is the logical order of text bits in the PDF (or may be the plain text layer is supposed to be better and ordered?). I don't know if libkopt uses that ordering to order how things are reflowed.

**Comment by jonnyl2** [comment]
Or let users manually override the ordering of highlights within a given page of a fixed-layout doc.
